### PR TITLE
fix(Dialog): reorder `Dismissable` so `focusout` works correctly

### PR DESCRIPTION
**Pull request by ArthurYidi**
Fixes: #2118

### CODE REVIEW

**Comment by TheEhsanSarshar** [comment]
Thanks for the PR!
it seems focus out handler is useful, as also mentioned in the Radix
https://github.com/radix-ui/primitives/blob/main/packages/react/focus-scope/src/FocusScope.tsx#L91

**Comment by ArthurYidi** [comment]
@TheEhsanSarshar yeah! Making some changes https://discord.com/channels/909986013848412191/994937898908516362/1198416664128147546

File: packages/dialog/src/Dialog.tsx [hunk 1]
```diff
@@ -584,28 +585,29 @@ const DialogContentImpl = React.forwardRef<TamaguiElement, DialogContentImplProp
...
-            ref={composedRefs}
-            onDismiss={() => context.onOpenChange(false)}
+            onUnmountAutoFocus={onCloseAutoFocus}
           >
+
             {contents}
-          </Dismissable>
-        </FocusScope>
+          </FocusScope>
+        </Dismissable>
```
Comment (ArthurYidi) [inline packages/dialog/src/Dialog.tsx:610]: Switches the order of `FocusScope` <-> `Dismissable`. That way `FocuScope`'s container is the dialog.

File: packages/dialog/src/Dialog.tsx [hunk 2]
```diff
@@ -460,6 +460,7 @@ const DialogContentModal = React.forwardRef<TamaguiElement, DialogContentTypePro
         onFocusOutside={composeEventHandlers(props.onFocusOutside, (event) =>
           event.preventDefault()
         )}
+        outlineStyle='none'
```
Comment (ArthurYidi) [inline packages/dialog/src/Dialog.tsx:463]: disable `outline` for the overall dialog (same approach as Adobe's spectrum aria)

**Comment by natew** [comment]
Thanks, tested it out and seems fine, but I do remember we've switched orders on these two before at some point. Couldn't find it, will test this out a bit more before releasing.

**Comment by natew** [comment]
Merged on another PR

**Pull request closed without merging**